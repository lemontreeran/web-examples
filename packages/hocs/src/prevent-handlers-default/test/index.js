import React from 'react'
import { mount } from 'enzyme'

import preventHandlersDefault from '../src/'

const Target = () => null

describe('preventHandlersDefault', () => {
  it('should pass props through if there are no arguments', () => {
    const EnhancedTarget = preventHandlersDefault()(Target)
    const wrapper = mount(
      <EnhancedTarget a={1} b={2} />
    )

    expect(wrapper.find(Target)).toMatchSnapshot()
  })

  it('should not wrap non-existing handler', () => {
    const EnhancedTarget = preventHandlersDefault('testHandler')(Target)
    const wrapper = mount(
      <EnhancedTarget />
    )

    expect(wrapper.find(Target)).toMatchSnapshot()
  })

  it('should call `preventDefault` from first argument if it has been passed to handler', () => {
    const mockPreventDefault = jest.fn()
    const testHandler = () => {}
    const EnhancedTarget = preventHandlersDefault('testHandler')(Target)
    const wrapper = mount(
      <EnhancedTarget testHandler={testHandler} />
    )

    wrapper.find(Target).prop('testHandler')({
      preventDefault: mockPreventDefault
    })
    expect(mockPreventDefault.mock.calls).toMatchSnapshot()
  })

  it('should not fail if there is no `preventDefault` in first handler argument', () => {
    const mockTestHandler = jest.fn()
    const EnhancedTarget = preventHandlersDefault('testHandler')(Target)
    const wrapper = mount(
      <EnhancedTarget testHandler={mockTestHandler} />
    )

    wrapper.find(Target).prop('testHandler')()
    expect(mockTestHandler.mock.calls).toMatchSnapshot()
  })

  describe('display name', () => {
    const origNodeEnv = process.env.NODE_ENV

    afterAll(() => {
      process.env.NODE_ENV = origNodeEnv
    })

    it('should wrap display name in non-production env', () => {
      process.env.NODE_ENV = 'test'

      const EnhancedTarget = preventHandlersDefault('testHandler')(Target)
      const testHandler = () => {}
      const wrapper = mount(
        <EnhancedTarget testHandler={testHandler} />
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('should not wrap display name in production env', () => {
      process.env.NODE_ENV = 'production'

      const EnhancedTarget = preventHandlersDefault('testHandler')(Target)
      const testHandler = () => {}
      const wrapper = mount(
        <EnhancedTarget testHandler={testHandler} />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })
})
