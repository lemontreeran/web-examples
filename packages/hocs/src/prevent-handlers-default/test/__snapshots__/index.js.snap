// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preventHandlersDefault display name should not wrap display name in production env 1`] = `
<PreventHandlersDefault
  testHandler={[Function]}
>
  <Target
    testHandler={[Function]}
  />
</PreventHandlersDefault>
`;

exports[`preventHandlersDefault display name should wrap display name in non-production env 1`] = `
<preventHandlersDefault(Target)
  testHandler={[Function]}
>
  <Target
    testHandler={[Function]}
  />
</preventHandlersDefault(Target)>
`;

exports[`preventHandlersDefault should call \`preventDefault\` from first argument if it has been passed to handler 1`] = `
Array [
  Array [],
]
`;

exports[`preventHandlersDefault should not fail if there is no \`preventDefault\` in first handler argument 1`] = `
Array [
  Array [
    undefined,
  ],
]
`;

exports[`preventHandlersDefault should not wrap non-existing handler 1`] = `<Target />`;

exports[`preventHandlersDefault should pass props through if there are no arguments 1`] = `
<Target
  a={1}
  b={2}
/>
`;
